modules/dxp/apps/commerce-demo-pack/commerce-theme-minium/commerce-theme-minium-full-site-initializer/src/main/resources/com/liferay/commerce/theme/minium/full/site/initializer/internal/dependencies/journal_articles/MINIUM_PROMO.ftl

```diff
@@ -1,31 +1,41 @@
-<script ${nonceAttribute}>
-	function getRelativeURL() {
-		Liferay.ctaURL = "${CTA.getData()}";
-
-		var currentRelativeURL = themeDisplay.getLayoutRelativeURL();
-
-		var relativeURLParts = currentRelativeURL.split('/');
-
-		Liferay.ctaURL = Liferay.ctaURL.replace("null", relativeURLParts[2]);
-
-		window.location.href = Liferay.ctaURL;
+<style ${nonceAttribute}>
+	.minium-promo-background-image{
+		background-image: linear-gradient(180deg, rgba(47, 51, 59, 0) 0%, #2F333B 100%), url(${ContentImage.getData()}); border-radius: 8px;
 	}
-</script>
+</style>
 
 <div class="banner-content-preview">
-	<div class="banner-content-section" style="background-image: linear-gradient(180deg, rgba(47, 51, 59, 0) 0%, #2F333B 100%), url(${ContentImage.getData()}); border-radius: 8px;">
+	<div class="banner-content-section minium-promo-background-image">
 		<div class="banner-content-container">
 			<h1 class="content-preview-title minium-h1">
 				<#if (Heading.getData())??>
 					${Heading.getData()}
 				</#if>
 			</h1>
 
-			<a class="minium-cta-link" onclick="getRelativeURL()">
+			<a class="minium-cta-link" href="#">
 				<button class="minium-cta-button">
 					<span class="minium-button-text">Discover</span>
 				</button>
 			</a>
 		</div>
+
+		<script ${nonceAttribute}>
+			const ctaLink = document.querySelector('.minium-cta-link');
+
+			function getRelativeURL() {
+				Liferay.ctaURL = "${CTA.getData()}";
+
+				var canonicalURL = themeDisplay.getCanonicalURL();
+
+				canonicalURL = canonicalURL.split('/dashboard')[0];
```

Comment:
The widget could be placed in a page which is not `dashboard`, hence the canonicalURL would be different and the split would fail.

Since we decided to keep the JS handler and simply move it below the element to prevent DOM null's, you can keep the previous logic.